tracer/build/_build/WindowsBuildFinishedNotificationAttribute.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+#if IS_WINDOWS
+using Microsoft.Toolkit.Uwp.Notifications;
+#endif
+using NuGet.Packaging;
+using Nuke.Common;
+using Nuke.Common.Execution;
+
+public class WindowsBuildFinishedNotificationAttribute : BuildExtensionAttributeBase, IOnBuildFinished
+{
+    private static readonly string AppData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+    private static readonly string BuildImages = Path.Combine(AppData, "build-images");
+    private static readonly string SucceedImage = Path.Combine(BuildImages, "build-succeed.jpg");
+    private static readonly string FailedImage = Path.Combine(BuildImages, "build-failed.jpg");
+    private static readonly string LogoImage = Path.Combine(BuildImages, "dd_logo.png");
+
+    public void OnBuildFinished(NukeBuild build)
+    {
+#if IS_WINDOWS
+        if (!(bool.TryParse(Environment.GetEnvironmentVariable("NUKE_NOTIFY"), out var notify)
+            && notify))
+        {
+            return;
+        }
+
+        CreateLocalImages();
+
+        var message =
+            build.IsSuccessful
+                ? $"Build succeeded on {DateTime.Now.ToString(CultureInfo.CurrentCulture)}. ＼（＾ᴗ＾）／"
+                : $"Build failed on {DateTime.Now.ToString(CultureInfo.CurrentCulture)}. (╯°□°）╯︵ ┻━┻";
+
+        var image =
+            build.IsSuccessful
+                ? SucceedImage
+                : FailedImage;
+
+        new ToastContentBuilder()
+            .AddInlineImage(new Uri($"file:///{image}"))
+            .AddAppLogoOverride(new Uri($"file:///{LogoImage}"), ToastGenericAppLogoCrop.Circle)
+            .AddText(message)
+            .Show();
+#endif
+    }
+
+    private static void CreateLocalImages()
+    {
+        if (!Directory.Exists(BuildImages))
+        {
+            Directory.CreateDirectory(BuildImages);
+        }
+
+        var urlBase = "https://github.com/robertpi/build-images/blob/main/";
+        var tasks = new []
+        {
+            DownloadAndCache(urlBase + "build-succeed.jpg?raw=true", SucceedImage),
+            DownloadAndCache(urlBase + "build-failed.jpg?raw=true", FailedImage),
+            DownloadAndCache(urlBase + "dd_logo.jpg?raw=true", LogoImage),
+        };
```

Comment:
Personally I would prefer if we stored them on the current repo (or at least a repo owned by datadog)

Reply:
It's easier if it's a public url, then no authentication is needed.

If it's this repo, then it get polluted with the image files and requires a PR to update them.

If it's a public repo owned by Datadog then it would need to comply to the open source standards.

I agree that it's a security issue having a url not owned by Datadog hardcoded, so I changed it so the user must choose the url, It falls back to a notification without images if no url is provided.